File: Components/DevExpressVCL/Demos/ExpressVerticalGrid/CBuilder/Common/DemoBasicAbout.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "DemoBasicAbout.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxControls"
#pragma link "cxClasses"
#pragma link "cxLookAndFeels"
#pragma link "cxGraphics"
#pragma link "cxLookAndFeelPainters"
#pragma link "cxButtons"
#pragma link "cxEdit"
#pragma link "cxImage"
#pragma resource "*.dfm"
TDemoBasicAboutForm *DemoBasicAboutForm;
//---------------------------------------------------------------------------
__fastcall TDemoBasicAboutForm::TDemoBasicAboutForm(TComponent* Owner)
  : TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicAboutForm::btnOKClick(TObject *Sender)
{
  Close();
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicAboutForm::FormClose(TObject *Sender,
  TCloseAction &Action)
{
  Action = caFree;
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicAboutForm::FormDestroy(TObject *Sender)
{
  DemoBasicAboutForm = NULL;
}
//---------------------------------------------------------------------------

File: Components/DevExpressVCL/Demos/ExpressVerticalGrid/Delphi/Common/DemoBasicAbout.pas
unit DemoBasicAbout;

interface

uses
  Windows, Messages, Classes, Graphics, Controls, Forms,
  Dialogs, StdCtrls, ExtCtrls, jpeg, ComCtrls, cxLookAndFeelPainters,
  cxButtons, cxControls, cxContainer, cxEdit, cxImage;

type
  TcxGetDemoGetAboutEvent = procedure(AAboutText: TStrings) of object;
  TcxGetDemoGetCaptionEvent = procedure(var ACaption: string) of object;

  TDemoBasicAboutForm = class(TForm)
    lbDemoName: TLabel;
    lbCopyright: TLabel;
    bvBottom: TBevel;
    lbCompanyName: TLabel;
    reDemoInfo: TRichEdit;
    btnOK: TcxButton;
    cxImage: TcxImage;
    procedure btnOKClick(Sender: TObject);
    procedure FormClose(Sender: TObject; var Action: TCloseAction);
    procedure FormDestroy(Sender: TObject);
  end;

var
  DemoBasicAboutForm: TDemoBasicAboutForm;

implementation

{$R *.dfm}

procedure TDemoBasicAboutForm.btnOKClick(Sender: TObject);
begin
  Close;
end;

procedure TDemoBasicAboutForm.FormClose(Sender: TObject;
  var Action: TCloseAction);
begin
  Action := caFree;
end;

procedure TDemoBasicAboutForm.FormDestroy(Sender: TObject);
begin
  DemoBasicAboutForm := nil;
end;

end.
